ci-operator/step-registry/ibmcloud/provision/dedicatedhost/static/ibmcloud-provision-dedicatedhost-static-commands.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#pre create the dedicated hosts based on the ${SHARED_DIR}/dedicated_host.yaml and update "${SHARED_DIR}/dedicated_host.yaml
+#save the dedicated host name for master nodes and worker nodes in "${SHARED_DIR}/dedicated_host"
+#save the resource group name of the dedicated host group in "${SHARED_DIR}/ibmcloud_resource_group_dhg"
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+## create the dedicated host group and dedicated host, if succeed, save the dedicated host name to file"
+## input parameter: the resource group of the dhg, dh profile, zone, name of the dh and the file which save the created dh name.
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4 file=$5
+    local tmp family class dhg status
+    tmp=$(${IBMCLOUD_CLI} is dedicated-host-profile $dProfile --output JSON | jq -r '.family +","+ .class')
+    family=${tmp%,*}
+    class=${tmp#*,}
+    echo "family:" $family " class:" $class
+    dhg="${dhName}g"
+    run_command "${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg}"
+    
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
+    
+    status=$(waitingStatus ${dhName})
+    echo "${dhName} status: ${status}"
+    run_command "${IBMCLOUD_CLI} is dedicated-host ${dhName}"
+
+    if [[ "${status}" != "stable" ]]; then
+        echo "ERROR: fail to create the dedicated host with profile $dProfile on zone: $zone resouce group:$rg"
+        echo "delete the dh ${dhName}, which status is ${status} ..."
+        run_command "${IBMCLOUD_CLI} is dhd ${dhName} -f"
```

Comment:
Why not leave the deprovision step to `ibmcloud-deprovision-dedicatedhost-static` ?

Reply:
The normal logic should be like:
1. provision DH failed -> the whole install failed, the prow ci job exit immediately -> `ibmcloud-deprovision-dedicatedhost-static` help clean the leftover.
2. provision DH passed -> the whole testing completed -> `ibmcloud-deprovision-dedicatedhost-static` help clean the created DH.

That is why I commented [there](https://github.com/openshift/release/pull/43123#discussion_r1354842740). Whatever DH creation failed or passed, always save the DH info in `dh_file`.